### PR TITLE
fix: refactor the workflow execution component to reduce repaint/rend…

### LINKED ISSUE #2025: [🐛 Bug]: Adding filter to workflow execution table cause refresh of the page
https://github.com/user-attachments/assets/b5562517-104d-43c6-994f-c59ff40d4515

### PR DESCRIPTION
**Pull request by rajeshj11**
…er the entire page when filters were applied and some minor style changes

Closes #2025

## 📑 Description

 refactor the workflow execution component to reduce repaint/re-rendering of entire page

## ✅ Checks

- [X] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [X] All the tests have passed

### CODE REVIEW

**Comment by rajeshj11** [comment]
@shahargl @talboren please review

**Comment by rajeshj11** [comment]
@Kiryous can you please review?

File: keep-ui/app/workflows/[workflow_id]/executions.tsx [hunk 1]
```diff
@@ -153,98 +147,178 @@ export default function WorkflowDetailPage({
...
+export default function WorkflowDetailPage({
+  params,
+}: {
+  params: { workflow_id: string };
+}) {
+  const router = useRouter();
+  const { data: session, status } = useSession();
+  const [navlink, setNavLink] = useState("overview");
+
+  const apiUrl = getApiURL();
```
Comment (Kiryous) [inline keep-ui/app/workflows/[workflow_id]/executions.tsx:251]:
```suggestion
  const apiUrl = useApiUrl();
```
Now you need to use this client hook to get API url. Please import it.

File: keep/api/routes/workflows.py [hunk 2]
```diff
@@ -491,6 +491,50 @@ def get_raw_workflow_by_id(
             )
         },
     )
+    
+@router.get("/{workflow_id}/data", description="Get workflow by ID")
```
Comment (Kiryous) [inline keep/api/routes/workflows.py:495]:
```suggestion
@router.get("/{workflow_id}", description="Get workflow by ID")
```

Let's use the route "/{workflow_id}" for general info about workflow and change current "/{workflow_id}" to "/{workflow_id}/runs"

File: keep-ui/app/workflows/[workflow_id]/executions.tsx [hunk 3]
```diff
@@ -85,14 +88,7 @@ interface Pagination {
...
 
-export default function WorkflowDetailPage({
-  params,
-}: {
-  params: { workflow_id: string };
-}) {
-  const router = useRouter();
-  const { data: session, status, update } = useSession();
-
+const OverViewContent = ({ workflow_id }: { workflow_id: string }) => {
```
Comment (Kiryous) [inline keep-ui/app/workflows/[workflow_id]/executions.tsx:91]: Let's rename this component and move it to its own file, `keep-ui/app/workflows/[workflow_id]/ui/WorkflowOverview.tsx
```suggestion
const WorkflowOverview = ({ workflow_id }: { workflow_id: string }) => {
```
`

File: keep/api/routes/workflows.py [hunk 4]
```diff
@@ -491,6 +491,50 @@ def get_raw_workflow_by_id(
...
+            providers=[],
+            triggers=[],
+            workflow_raw=final_workflow_raw,
+            revision=workflow.revision,
+            last_updated=workflow.last_updated,
+            disabled=workflow.is_disabled,
+            provisioned=workflow.provisioned,
+        )
+        return workflow_dto
+    except Exception as e:
```
Comment (Kiryous) [inline keep/api/routes/workflows.py:534]: Do we know any specific exceptions to expect here?

File: keep-ui/app/workflows/builder/builder-modal.tsx [hunk 5]
```diff
@@ -11,13 +11,15 @@ import { downloadFileFromString } from "./utils";
...
-  id?:string
+  id?: string;
+  hideClose?: boolean;
 }
 
 export default function BuilderModalContent({
   closeModal,
   compiledAlert,
   id,
+  hideClose,
```
Comment (Kiryous) [inline keep-ui/app/workflows/builder/builder-modal.tsx:22]: Let's use cleaner name 
```suggestion
  hideCloseButton,
```

**Review by Kiryous — CHANGES_REQUESTED** [review]
Hey Rajesh, thank you for your contribution! I left a few comments, please consider fixing them. Sorry for a delayed response.

File: keep/api/routes/workflows.py [hunk 6]
```diff
@@ -563,7 +563,7 @@ def get_workflow_executions_by_alert_fingerprint(
     ]
 
 
-@router.get("/{workflow_id}", description="Get workflow executions by ID")
+@router.get("/{workflow_id}/logs", description="Get workflow executions by ID")
```
[the remark this answers is not in the document]
Comment (Kiryous) [reply keep/api/routes/workflows.py:566]: No, it won't conflict, because the route "/{workflow_id}/runs" would be resolved earlier. Please change.

**Review by Kiryous — APPROVED** [review]
LGTM! Thanks for the contribution, Rajesh!

**Pull request merged**